examples/src/main/java/com/mapbox/navigation/examples/core/ReplayEngineActivity.kt

```diff
@@ -0,0 +1,266 @@
+
+        mapboxNavigation = MapboxNavigation(
+                applicationContext,
+                Utils.getMapboxAccessToken(this),
+                mapboxNavigationOptions,
+                locationEngine = replayRouteLocationEngine
+        )
+
+        initListeners()
+    }
+
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            locationEngine = LocationEngineProvider.getBestLocationEngine(this)
+            initLocationComponent(style, mapboxMap)
+            navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap).also {
+                it.addProgressChangeListener(mapboxNavigation!!)
+                mapInstanceState?.let { state ->
+                    it.restoreFrom(state)
+                }
+            }
+        }
+        mapboxMap.addOnMapLongClickListener { latLng ->
+            mapboxMap.locationComponent.lastKnownLocation?.let { originLocation ->
+                mapboxNavigation?.requestRoutes(
+                        RouteOptions.builder().applyDefaultParams()
+                                .accessToken(Utils.getMapboxAccessToken(applicationContext))
+                                .coordinates(originLocation.toPoint(), null, latLng.toPoint())
+                                .alternatives(true)
+                                .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                                .build(),
+                        routesReqCallback
+                )
+            }
+            true
+        }
+        locationComponent = mapboxMap.locationComponent
+    }
+
+    @SuppressLint("RestrictedApi")
+    fun initLocationComponent(loadedMapStyle: Style, mapboxMap: MapboxMap) {
+        when (PermissionsManager.areLocationPermissionsGranted(this)) {
+            true -> {
+                mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+                mapboxMap.locationComponent.let { locationComponent ->
+                    val locationComponentActivationOptions =
+                            LocationComponentActivationOptions.builder(this, loadedMapStyle)
+                                    // .useDefaultLocationEngine(true)
+                                    .build()
+
+                    locationComponent.activateLocationComponent(locationComponentActivationOptions)
+                    locationComponent.isLocationComponentEnabled = true
+                    locationComponent.cameraMode = CameraMode.TRACKING
+                    locationComponent.renderMode = RenderMode.COMPASS
+
+                    initLocationEngine()
+                }
+            }
+            false -> {
+                permissionsManager =
+                        PermissionsManager(this).also { it.requestLocationPermissions(this) }
+            }
+        }
+    }
+
+    fun initLocationEngine() {
+        val requestLocationUpdateRequest =
+                LocationEngineRequest.Builder(DEFAULT_INTERVAL_IN_MILLISECONDS)
```

Comment:
it'd be nice to have a default location engine config that's recommended for navigation